File: tmp/fix_track.py
import json

# Adjust this file path as per the environment
file_path = "d:/DATAATAR/UBIG/balap/src/lib/trackAssets.json"

with open(file_path, "r") as f:
    data = json.load(f)

# Sort by Z to process in order
data.sort(key=lambda x: x['z'])

# Scaling factor to give more space
# If buildings were at 50, 60, 90, 100...
# Scaling by 2.5x makes them 125, 150, 225, 250...
# This doubles the gap between building clusters.
FACTOR = 2.5

new_data = []
for item in data:
    item['z'] = round(item['z'] * FACTOR)
    new_data.append(item)

with open(file_path, "w") as f:
    json.dump(new_data, f, indent=4)

print(f"Track assets scaled by {FACTOR}x")

File: tmp/fix_track.js
const fs = require('fs');
const path = 'd:/DATAATAR/UBIG/balap/src/lib/trackAssets.json';

const data = JSON.parse(fs.readFileSync(path, 'utf8'));

// Double the distance between objects
// And apply extra gap if objects are buildings that might overlap due to their size
data.sort((a,b) => a.z - b.z);

const FACTOR = 2.5;
data.forEach(item => {
    item.z = Math.round(item.z * FACTOR);
});

fs.writeFileSync(path, JSON.stringify(data, null, 4));
console.log('Track assets scaled successfully 2.5x');
